### PR TITLE
ddl: add metrics and clean up

### PR DESCRIPTION
**Pull request by zimulala**
## What have you changed? (mandatory)
Add the type of DDL worker to metrics.
Move `generalWorker` to ddl_test.go
Use the internal auto-increment ID instead of passing the ID.

## What is the type of the changes? (mandatory)

- New feature (non-breaking change which adds functionality)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
Yes.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No.

## Does this PR affect tidb-ansible update? (mandatory)
No.

## Does this PR need to be added to the release notes? (mandatory)
No.

### CODE REVIEW

File: ddl/ddl_worker.go [hunk 1]
```diff
@@ -32,8 +33,12 @@ import (
 	"golang.org/x/net/context"
 )
 
-// RunWorker indicates if this TiDB server starts DDL worker and can run DDL job.
-var RunWorker = true
+var (
+	// RunWorker indicates if this TiDB server starts DDL worker and can run DDL job.
+	RunWorker = true
+	// ddlWorkerID is used for generating the next DDL worker ID.
+	ddlWorkerID = int32(0)
```
Comment (ngaut) [inline ddl/ddl_worker.go]: How about use int64?

Comment (zimulala) [reply ddl/ddl_worker.go]: why? The number of workers is two now,  and there will be no more than ten in the future.

Comment (ngaut) [reply ddl/ddl_worker.go]: The origin type of id is "int", which is 64 bit actually. When i review the code, i was wondering why changed to int32.

**Comment by zimulala** [comment]
@coocood 
Why?

File: ddl/ddl_worker.go (hunk 1 above)
Comment (zimulala) [reply ddl/ddl_worker.go]: @ngaut 
Because I want to use "atomic".

Comment (zimulala) [reply ddl/ddl_worker.go]: @coocood 
I use "atomic" to generate the DDL worker ID. And it's only used in log and metric. I think it's OK.

**Pull request merged**